app/customer.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import annotations
+
+from decimal import Decimal
+from typing import List
+
+from app.car import Car
+from app.shop import Shop
+
+
+class Customer:
+    def __init__(
+            self,
+            name: str,
+            product_cart: dict,
+            location: list,
+            money: int,
+            car: Car
+    ) -> None:
+        self.name = name
+        self.product_cart = product_cart
+        self.location = location
+        self.home_location = location
+        self.money = money
+        self.car = car
+
+    def find_optimal_shop(
+            self,
+            shops: List[Shop],
+            fuel_price: float
+    ) -> list[Shop | str | int]:
+        print(f"{self.name} has {self.money} dollars")
+        optimal_shop = [shops[0], "", 10 ** 6]

```

Comment:
Why did you choose such a structure? 
To achieve better readability and make you code more understandable, you can use dictionary with shop (or shop name) as key and total cost as a value

Then you can call min() function to determine the shop with the smallest total_cost